torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -8935,20 +8982,89 @@ def _test_hook_pickling(self, hook, hook_state):
+        @skip_if_lt_x_gpu(2)
+        def test_ddp_hook_pickling_powerSGD_logs(self):
 
             hook = powerSGD.powerSGD_hook
             powersgd_state = powerSGD.PowerSGDState(
                 process_group=None,
                 matrix_approximation_rank=1,
                 start_powerSGD_iter=4,
             )
-            self._test_hook_pickling(hook, powersgd_state)
+            restore_msg = "NOTE: Process group will be set to a default group (i.e. the world size).\n"\
+                "If a different group is desired, please set `self.process_group` after PowerSGD state is loaded."
+
+            self._test_hook_pickling_logs(hook, powersgd_state, restore_msg=restore_msg)
+
+        @sandcastle_skip_if(
+            BACKEND not in DistTestCases.backend_feature["cuda"],
+            f"The {BACKEND} backend does not support DDP communication hook on CUDA devices"
+        )
+        @skip_if_lt_x_gpu(2)
+        def test_ddp_hook_pickling_powerSGD_parity(self):
+
+            hook = powerSGD.powerSGD_hook
+            powersgd_state = powerSGD.PowerSGDState(
+                process_group=None,
+                matrix_approximation_rank=1,
+                start_powerSGD_iter=4,
+            )
+            self._test_hook_pickling_parity(hook, powersgd_state)
+
+        @sandcastle_skip_if(
+            BACKEND not in DistTestCases.backend_feature["cuda"],
+            f"The {BACKEND} backend does not support DDP communication hook on CUDA devices"
+        )
+        @sandcastle_skip_if(
+            BACKEND not in DistTestCases.backend_feature["subgroup"],
+            f"The {BACKEND} backend does not support creating subgroups on CUDA devices"
+        )
+        @skip_if_lt_x_gpu(2)
+        def test_ddp_hook_pickling_postlocal_SGD_logs(self):
+
+            hook = post_localSGD.post_localSGD_hook
+            postlocalsgd_state = post_localSGD.PostLocalSGDState(
+                process_group=None,
+                subgroup=None,
+                start_localSGD_iter=0
+            )
+            restore_msg = "NOTE: Process group will be set to a default group (i.e. the world size).\n"\
+                "Subgroup will be set to a default subgroup (i.e. an intra-node subgroup)\n"\
+                "If a different group is desired, please set `self.process_group` after `PostLocalSGD`'s state is loaded."
+            self._test_hook_pickling_logs(hook, postlocalsgd_state, restore_msg=restore_msg)
+
+        @sandcastle_skip_if(
+            BACKEND not in DistTestCases.backend_feature["cuda"],
+            f"The {BACKEND} backend does not support DDP communication hook on CUDA devices"
+        )
+        @sandcastle_skip_if(
+            BACKEND not in DistTestCases.backend_feature["subgroup"],
+            f"The {BACKEND} backend does not support creating subgroups on CUDA devices"
+        )
+        @skip_if_lt_x_gpu(2)
+        def test_ddp_hook_pickling_postlocal_SGD_parity(self):
+
```

Comment:
nit: do we want this extra newline?

Reply:
well, emptiness always brings nothing to anything, so I agree, we don't want it